core/src/main/java/com/homeaway/streamplatform/streamregistry/resource/ClusterResource.java

```diff
@@ -0,0 +1,177 @@
+            log.error("Error upserting cluster.", e);
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(new ErrorMessage(Response.Status.BAD_REQUEST.getStatusCode(),
+                    "Error upserting cluster.",
+                    e.getCause() != null ? e.getMessage() + ". " + e.getCause().getMessage() : e.getMessage()))
+                .build();
+        } catch (Exception e) {
+            log.error("Error upserting cluster.", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                .entity(new ErrorMessage(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                    "Error upserting cluster.",
+                    e.getMessage()))
+                .build();
+        }
+    }
+
+    @GET
+    @Path("/")
+    @ApiOperation(
+        value = "Get All Clusters",
+        notes = "Gets all the clusters",
+        tags = "clusters")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Returns Cluster information"),
+        @ApiResponse(code = 500, message = "Error Occurred while getting data"),
+        @ApiResponse(code = 404, message = "Cluster not found") })
+    @Produces(MediaType.APPLICATION_JSON)
+    @Timed
+    public Response getAllClusters(@ApiParam(value = "vpc") @QueryParam("vpc") String vpc,
+        @ApiParam(value = "env") @QueryParam("env") String env,
+        @ApiParam(value = "hint") @QueryParam("hint") String hint,
+        @ApiParam(value = "type") @QueryParam("type") String type) {
+        try {
+            Map<ClusterKey, ClusterValue> clusterMap = clusterDao.getAllClusters();
+
+            if(vpc != null && !vpc.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getVpc().equalsIgnoreCase(vpc))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            if(env != null && !env.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getEnv().equalsIgnoreCase(env))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            if(hint != null && !hint.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getHint().equalsIgnoreCase(hint))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            if(type != null && !type.isEmpty()) {
+                clusterMap = clusterMap.entrySet().stream()
+                    .filter(e -> e.getKey().getType().equalsIgnoreCase(type))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+
+            return Response.status(200).entity(clusterMap).build();
+        } catch (Exception e) {
+            log.error("Error getting cluster details.", e);
```

Comment:
lets catch any runtime exception if any.

if it is checked exceptions, lets catch the particular ones and take a action on it